File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            cmp = target-nums[i]
            if cmp in d.keys():
                return [d[cmp], i]
            else:
                d[nums[i]] = i
        return []

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        for(int i=0; i<nums.length; i++){
            int cmp = target - nums[i];
            if(hm.containsKey(cmp)){
                return new int[] {hm.get(cmp), i};
            }
            else{
                hm.put(nums[i],i);   
            }
        }
        return new int[] {};
    }
}